helm-prereqs/setup.sh

```
@@ -1812,6 +1819,50 @@ _KC_ENABLED="$(grep -A5 'keycloak:' "${SCRIPT_DIR}/values/nico-rest.yaml" \
     echo "=== [7d/7] Keycloak ==="
+
+    # helm-prereqs/values.yaml::keycloak.enabled — whether Keycloak's DB lives
+    # on nico-pg-cluster (opt-in) instead of the legacy postgres.postgres
+    # StatefulSet (default). Distinct from nico-rest.yaml's keycloak.enabled
+    # checked above, which gates whether Keycloak is deployed at all.
+    _KC_DB_CONSOLIDATED="$(grep -A3 '^keycloak:' "${SCRIPT_DIR}/values.yaml" \
+        | grep 'enabled:' | head -1 | awk '{print $2}' || echo "false")"
+    # Same helm-prereqs/values.yaml::keycloak block — this is the namespace
+    # eso-external-secrets.yaml's nico-keycloak-db-eso actually targets, so
+    # KEYCLOAK_NS (read by keycloak/setup.sh, default nico-rest) must match it.
+    export KEYCLOAK_NS="$(grep -A3 '^keycloak:' "${SCRIPT_DIR}/values.yaml" \
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)